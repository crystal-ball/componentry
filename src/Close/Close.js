import React from 'react'
import simpleComponent from '../simple-component-factory'

export const closeBase = {
  'aria-label': 'close',
  'as': 'button',
  'elemClassName': 'close',
  'type': 'button',
  'children': (
    <svg className='icon icon-close font' role='img'>
      <use href='#close' xlinkHref='#close' />
    </svg>
  ),
}

/**
 * [Close component üìù](https://componentry.design/components/close)
 */
export default simpleComponent('Close', closeBase)
