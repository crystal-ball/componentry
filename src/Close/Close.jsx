import React from 'react'

import { BaseButton } from '../Button/Button'
import elementFactory from '../component-factories/element-factory'

const Close = elementFactory({
  name: 'Close',
  classes: 'btn-close',
  tag: BaseButton,
  children: (
    <svg className="icon close font" role="img" aria-label="close">
      <use href="#close" />
    </svg>
  ),
})

export default Close
