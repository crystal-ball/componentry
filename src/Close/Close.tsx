/* eslint-disable @typescript-eslint/no-empty-interface */
import React from 'react'
import { ComponentBaseProps } from '../utils/types'
import { staticComponent } from '../utils/static-component-builder'

export interface CloseProps extends ComponentBaseProps<'button'> {}

export const closeBase: CloseProps = {
  'aria-label': 'close',
  'as': 'button',
  'type': 'button',
  'children': (
    <svg className='üÖ≤-icon icon-close font-variant' role='img'>
      <use href='#close' xlinkHref='#close' />
    </svg>
  ),
}

/**
 * [Close component üìù](https://componentry.design/components/close)
 * @experimental
 */
export const Close = staticComponent<CloseProps>('Close', closeBase)
