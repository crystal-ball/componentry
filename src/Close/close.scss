/* componentry/src/close */

// The btn-close allows for targeted styles for the close buttons. The .icon.close
// can also be styled for close buttons only using that as a target
// ℹ️ The background image styles for close icons is located in `/styles/componentry/icons`
.btn-close {
  @extend .btn;
  @extend .btn-anchor;

  color: $close-color;

  @include hover-focus {
    color: $close-color;
  }

  // Animate close icon opacity on hover
  .icon.close {
    opacity: 0.6;
    transition: $transition-fade;

    @include hover-focus {
      opacity: 1;
    }
  }
}

// By default Componentry provides a very simple set of icon classes with background
// images, but this is not the recommended system. Consumers should set up an SVG
// icon system and set $include-icons to false to use real SVGs for icons.
@if $include-icons {
  .icon.close {
    @extend .componentry-icon;

    background-image: str-replace(
      url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='#{$body-color}' viewBox='0 0 32 32'%3E%3Cpath d='M32,3.22,19.22,16,32,28.78,28.78,32,16,19.22,3.22,32,0,28.78,12.78,16,0,3.22,3.22,0,16,12.78,28.78,0Z'/%3E%3C/svg%3E"),
      '#',
      '%23'
    );
  }
}

// ℹ️ Do not import Bootstrap close styles:
// The Bootstrap close styles assume that you are using the &times; character for
// your close button, and includes some styles to bump the font size, adjust line
// line height and add a float.
//
// Componentry supports using an SVG icon so that applications can choose a close
// icon that fits their theme, as well as using flexbox on containers with close
// buttons instead of floats.
