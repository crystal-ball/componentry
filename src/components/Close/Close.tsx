import React from 'react'
import { type ClassValue } from 'clsx'
import { ComponentBaseProps } from '../../utils/types'
import { staticComponent } from '../../utils/static-component-builder'
import { Icon } from '../Icon/Icon'

export interface CloseProps extends ComponentBaseProps<'button'> {
  componentCx?: ClassValue
}

export const closeBase: CloseProps = {
  'aria-label': 'close',
  'as': 'button',
  'type': 'button',
  'componentCx': `üÖ≤Close-base`,
  'children': <Icon id='close' />,
}

/**
 * [Close component üìù](https://componentry.design/components/close)
 * @experimental
 */
export const Close = staticComponent<CloseProps>('Close', closeBase)
