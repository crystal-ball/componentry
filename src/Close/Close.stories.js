/* global module */

import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import Close from './Close'

storiesOf('Close', module).add('<Close />', () => <Close onClick={action('close')} />)
