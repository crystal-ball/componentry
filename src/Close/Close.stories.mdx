import { Close } from './Close'

<Meta title='Components/Close' component={Close} />

<DocsTitle title='Close' experimental />

Close component is used in the `Alert` and `Modal` components.

<Close onClick={console.log} />
