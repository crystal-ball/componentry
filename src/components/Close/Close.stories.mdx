import { ArgsTable, Meta, Story } from '@storybook/addon-docs'

import { Close } from './Close'

<Meta title='Components/Close' component={Close} />

# Close

The Close component wraps an instance of Icon to create an accessible close element for
the Alert and Modal components. The `C9Y-Close-base` class provides a target for custom
styling when needed.

<Story name='Close'>
  <Close onClick={console.log} />
</Story>
