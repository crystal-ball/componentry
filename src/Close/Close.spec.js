import React from 'react'
import { render } from '@testing-library/react'

import Close from './Close'
import elementTests from '../utils-test/element-tests'

describe('<Close />', () => {
  // Basic library element test suite
  elementTests(Close)
})

// Snapshots
// ---------------------------------------------------------------------------
describe('<Close /> snapshots', () => {
  it('renders correctly', () => {
    const { container } = render(<Close />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
