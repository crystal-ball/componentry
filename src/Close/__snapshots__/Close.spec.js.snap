// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Close /> snapshots renders correctly 1`] = `
<button
  aria-label="close"
  class="ðŸ…²-close"
  type="button"
>
  <svg
    class="ðŸ…²-icon icon-close font-variant"
    role="img"
  >
    <use
      href="#close"
      xlink:href="#close"
    />
  </svg>
</button>
`;
